docs/reference/ml/ml-shared.asciidoc

```diff
@@ -361,6 +361,18 @@ example, it can contain custom URL information as shown in
 {ml-docs}/ml-configuring-url.html[Adding custom URLs to {ml} results].
 end::custom-settings[]
 
+tag::daily-model-snapshot-retention-after-days[]
+Advanced configuration option. Specifies a number of days between 0 and the
+value of `model_snapshot_retention_days`. After this period of time, only one
+model snapshot per day is retained for this job. Age is calculated relative to
```

Comment:
> Age is calculated relative to the timestamp of the newest model snapshot...

Is this sentence needed?  If yes, it's unclear to me what this age affects.

Reply:
I think _something_ is needed.  Not sure how best to phrase it though.

Suppose you have the following model snapshots:

1. Tuesday 9:04 am
2. Monday 10:39 am
3. Sunday 9:03 am
4. Saturday 10:46 pm
5. Friday 8:02 pm
6. Friday 7:49 pm
7. Friday 8:59 am
8. Thursday 9:04 am

And suppose `model_snapshot_retention_days` is 4 and `daily_model_snapshot_retention_after_days` is 2.

Then we will delete all snapshots older than Friday 9:04 am (= Tuesday 9:04 am - 4 days).  So we will delete snapshots 7 and 8 by this rule.  And for snapshots at or after Friday 9:04 am (= Tuesday 9:04 am - 4 days) but before Sunday 9:04 am (= Tuesday 9:04 am - 2 days) we will keep the first per 24 hour period.  So we will keep snapshots 6 and 4 by this rule and delete snapshots 5 and 3.

You can see that this is all highly dependent on the timestamp of the latest model snapshot, because everything is measured relative to its timestamp.  So this is what "Age is calculated relative to the timestamp of the newest model snapshot" was trying to say.  Can you recommend a good way to explain it?  I will probably do this docs change in a followup PR though if the PR build goes green soon as I need to get the main code change merged before the 7.8 branch is split.

Reply:
I've created https://github.com/elastic/stack-docs/pull/1047 to update the Model snapshots (https://www.elastic.co/guide/en/machine-learning/current/ml-model-snapshots.html) page and can revisit the API details thereafter.

Reply:
Thanks Lisa!